python/spark_sklearn/test_utils.py

```diff
@@ -16,21 +16,32 @@
 from scipy.sparse import csr_matrix
 
 from pyspark import SparkContext
-from pyspark.sql import SQLContext
-from pyspark.mllib.linalg import Vectors
+from pyspark.sql import SparkSession
+from pyspark.ml.linalg import Vectors
 
-sc = None
+# Used as deocrator to have one PySpark SparkSession per fixture.
+def fixtureReuseSparkSession(cls):
+    setup = getattr(cls, 'setUpClass', None)
+    teardown = getattr(cls, 'tearDownClass', None)
+    def setUpClass(cls):
+        if setup: setup()
+        cls.spark = SparkSession.builder.master("local").appName("Unit Tests").getOrCreate()
+    def tearDownClass(cls):
+        if cls.spark:
+            cls.spark.stop()
+            SparkSession._instantiatedContext = None
```

Comment:
Need what? Spark reuse? For speed. Decorator? Because it's used quite a few times